drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -107,6 +116,23 @@ local function handle_battery_charge_level(driver, device, ib, response)
   end
 end
 
+local function handle_power_source_attribute_list(driver, device, ib, response)
+  local support_battery_percentage = false
+  for _, attr in ipairs(ib.data.elements) do
+    -- Re-profile the device if BatPercentRemaining (Attribute ID 0x0C) is present.
+    if attr.value == 0x0C then
+      support_battery_percentage = true
+    end
+  end
+  device:set_field(SUPPORT_BATTERY_PERCENTAGE, support_battery_percentage, {persist = true})
+  local profile_name = device:get_field(PROFILE_BASE_NAME)
+  if not device:get_field(SUPPORT_BATTERY_PERCENTAGE) then
+    profile_name = profile_name .. "-batteryLevel"
+  end
+  device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
```

Comment:
I don't think this needs to be included in the hub logs personally, but could be left in the driver logs.

Reply:
Updated as suggested!